client/network/src/block_request_handler.rs

```diff
@@ -192,19 +180,18 @@ impl<B: BlockT> BlockRequestHandler<B> {
 			support_multiple_justifications,
 		};
 
-		let mut reputation_changes = Vec::new();
+		let mut reputation_change = None;
 
 		match self.seen_requests.get_mut(&key) {
-			Some(SeenRequestsValue::First) => {},
-			Some(SeenRequestsValue::Fulfilled(ref mut requests)) => {
+			Some(requests) => {
```

Comment:
This is not correct. We should only start incremented after the first time we have fulfilled the request...